lib/fastlyPromise.js

```diff
@@ -145,4 +149,170 @@ FastlyPromise.prototype.purgeKey = function purgeKey(serviceId, surrogateKey, so
+ *
+ * @param serviceId
+ * @param [configVersionNumber] - Defaults to active config version.
+ */
+FastlyPromise.prototype.cloneConfigVersion = function cloneConfigVersion(serviceId, configVersionNumber) {
+  var self = this;
+
+  //Set default config version number to active version if not explicitly defined.
+  var configVersionNumberPromise = configVersionNumber ? Promise.resolve(configVersionNumber) : self.getActiveConfigVersion(serviceId).then(function(version){ return version.number });
+
+  return configVersionNumberPromise.then(function(configVersionNumber){
+    return self.request('PUT', self.endpoint + '/service/' + serviceId + '/version/' + configVersionNumber + '/clone');
+  });
+
+
+}
+
+/**
+ *
+ * Get a list of service VCLs by config version.
+ *
+ * @param serviceId
+ * @param configVersionNumber
+ */
+FastlyPromise.prototype.getAllVcl = function getAllVcl(serviceId, configVersionNumber) {
+
+  return this.request('GET', this.endpoint + '/service/' + serviceId + '/version/' + configVersionNumber + '/vcl');
+
+}
+
+/**
+ *
+ * Get a service VCL by config version and name.
+ *
+ * @param serviceId
+ * @param configVersionNumber
+ * @param vclName
+ */
+FastlyPromise.prototype.getVcl = function getVcl(serviceId, configVersionNumber, vclName) {
+
+  return this.request('GET', this.endpoint + '/service/' + serviceId + '/version/' + configVersionNumber + '/vcl/' + vclName);
+
+}
+
+/**
+ *
+ * Get the currently active VCL for a service.
+ *
+ * @param {string} serviceId
+ * @param {string} [configVersionNumber] - If no config version number is passed, defaults to active config version.
+ */
+FastlyPromise.prototype.getActiveVcl = function getActiveVcl(serviceId, configVersionNumber) {
+  var self = this;
+
+  //Set default config version number to active version if not explicitly defined.
+  var configVersionNumberPromise = configVersionNumber ? Promise.resolve(configVersionNumber) : self.getActiveConfigVersion(serviceId).then(function(version){ return version.number });
+
+  return configVersionNumberPromise
+    .then(function(configVersionNumber){
+    return self.getAllVcl(serviceId, configVersionNumber);
+  })
+    .then(function(vclList){
+
+      return vclList.find(function(vcl){
+        return vcl.main;
+      });
+
+    });
+
+}
+
+/**
+ *
+ * Upload new VCL to a service. Throws error if VCL with same name already exists.
+ *
+ * @param {string} serviceId
+ * @param {string} configVersionNumber - Service config version number.
+ * @param {string} vclName - Name of vcl file.
+ * @param {string} vclContent - The VCL content to be uploaded.
+ * @param {boolean} [setVclToMain=false] - Set the uploaded vcl to be the service's "main" vcl.
+ */
+FastlyPromise.prototype.uploadNewVcl = function uploadNewVcl(serviceId, configVersionNumber, vclName, vclContent, setVclToMain) {
+  var self = this;
+
+  setVclToMain = setVclToMain || false;
```

Comment:
Should probably be using ES6 default parameters for `setVclToMain` instead of this line?